projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts

```diff
@@ -99,9 +117,23 @@ export class LazyElementsLoaderService {
         script.type = 'module';
       }
       script.src = url;
-      script.onload = notifier.resolve;
+      script.onload = () => {
+        if (hooksConfig?.afterLoad) {
+          this.handleHook(hooksConfig.afterLoad, tag)
+            .then(notifier.resolve)
+            .catch(notifier.reject);
+        } else {
+          notifier.resolve();
+        }
+      };
       script.onerror = notifier.reject;
-      document.body.appendChild(script);
+      if (hooksConfig?.beforeLoad) {
```

Comment:
What happens if there is more than one hook? 

eg from global config and from element config ? 

Should be also possible to specify global hooks ? (in root config ? )

Reply:
Added support for root-level hooks. Service will try to run element-level hook if present and will fall back to root-level hook. I think that is the most flexible approach as that way hooks can be overridden for certain elements.

Reply:
@MagicCurlyHair sounds great!